src/s2i/bin/assemble

```diff
@@ -36,7 +36,15 @@ function install_tool() {
   {% endif %}
   VENV_DIR=$HOME/.local/venvs/$1
   virtualenv_bin "$VENV_DIR"
-  $VENV_DIR/bin/pip --isolated install -U $1$2  # Combines package name with [extras] if [extras] is defined as $2
+  # First, try to install the tool without --isolated which means that if you
+  # have your own PyPI mirror, it will take it from there. If this try fails, try it
+  # again with --isolated which ignores external pip settings (env vars, config file)
+  # and installs the tool from PyPI (needs internet connetion).
+  $VENV_DIR/bin/pip install -U $1$2  # Combines package name with [extras] if [extras] is defined as $2
+  if [ $? -ne 0 ]; then
+    echo "WARNING: Installation of $1 failed, trying again from official PyPI with pip --isolated"
+    $VENV_DIR/bin/pip --isolated install -U $1$2  # Combines package name with [extras] if [extras] is defined as $2
```

Comment:
```suggestion
  $VENV_DIR/bin/pip install -U $1$2  # Combines package name with [extras] or version specifier if is defined as $2```

Reply:
BTW, I'll adjust this one because the indentation should remain there.

Reply:
Oh, of course, my mistake.